Libraries/Image/RCTImageLoader.m

```diff
@@ -390,31 +401,6 @@ - (RCTImageLoaderCancellationBlock)loadImageOrDataWithURLRequest:(NSURLRequest *
       request = mutableRequest;
     }
 
-    // Check for cached response before reloading
-    // TODO: move URL cache out of RCTImageLoader into its own module
-    NSCachedURLResponse *cachedResponse = [_URLCache cachedResponseForRequest:request];
-
-    while (cachedResponse) {
-      if ([cachedResponse.response isKindOfClass:[NSHTTPURLResponse class]]) {
-        NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)cachedResponse.response;
-        if (httpResponse.statusCode == 301 || httpResponse.statusCode == 302) {
```

Comment:
And I finally found the [docs](https://developer.apple.com/library/ios/documentation/Foundation/Reference/NSURLSession_class/index.html) about this.


>If the response is an HTTP redirect response, the NSURLSession object calls the delegate’s URLSession:task:willPerformHTTPRedirection:newRequest:completionHandler: method. That delegate method calls the provided completion handler with either the provided NSURLRequest object (to follow the redirect), a new NSURLRequest object (to redirect to a different URL), or nil (to treat the redirect’s response body as a valid response and return it as the result).

> If you decide to follow the redirect, go back to step 2.

> **If the delegate doesn’t implement this method, the redirect is followed up to the maximum number of redirects.**